src/dirs.proj

```diff
@@ -19,11 +19,44 @@
     <Project Include="XMakeBuildEngine\UnitTestsPublicOM\Microsoft.Build.Engine.OM.UnitTests.csproj" />
   </ItemGroup>
 
+  <Import Project="dir.targets" />
   <Import Project="..\dir.traversal.targets" />
 
   <PropertyGroup>
+    <SkipVersionGeneration>true</SkipVersionGeneration>
+    <!-- Turn off the automated package generation in packages.targets:AddNuGetPackageVersionMetadataToNuspecs;
+         we'll use the one produced by NerdBank.GitVersioning instead. -->
+    <DoNotGeneratePackageVersion>true</DoNotGeneratePackageVersion>
+  </PropertyGroup>
+
+  <Import Project="$(ToolsDir)packages.targets" Condition="Exists('$(ToolsDir)packages.targets') and '$(ImportGetNuGetPackageVersions)' != 'false'" />
+
+  <PropertyGroup Condition="Exists('$(ToolsDir)packages.targets') and '$(ImportGetNuGetPackageVersions)' != 'false'">
+    <TraversalBuildDependsOn>
+      $(TraversalBuildDependsOn);
+      GetNuspecProperties;
+      BuildPackages;
+    </TraversalBuildDependsOn>
+    <!-- We use Microsoft.DotNet.Build tools to create our packages, but we want the version comes from 
+         NerdBack.GitVersioning. -->
+    <GetNuGetPackageVersionsDependsOn>SetPackageVersion</GetNuGetPackageVersionsDependsOn>
```

Comment:
@rainersigwald So `BeforeTargets="BuildPackages"` is enough to ensure `SetPackageVersion` is run, even if nothing depends on it directly?

There are two things I'm trying to achieve: get the package version from GitVersioning and pass that along to `BuildPackages` (by setting `PackageVersion` before `BuildPackages` runs), and create the `NuspecProperties` items, which also need the version from GitVersioning and also need to be created before `BuildPackages` runs.

I'll try your suggestion.